system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.hpp

```diff
@@ -0,0 +1,373 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef TOPIC_STATISTICS_COLLECTOR__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+#define TOPIC_STATISTICS_COLLECTOR__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+
+#include <chrono>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rcpputils/asserts.hpp"
+
+#include "libstatistics_collector/topic_statistics_collector/received_message_age.hpp"
+#include "libstatistics_collector/topic_statistics_collector/received_message_period.hpp"
+#include "libstatistics_collector/topic_statistics_collector/topic_statistics_collector.hpp"
+#include "system_metrics_collector/constants.hpp"
+#include "system_metrics_collector/metrics_message_publisher.hpp"
+
+
+namespace topic_statistics_collector
+{
+
+using libstatistics_collector::topic_statistics_collector::TopicStatisticsCollector;
+using libstatistics_collector::topic_statistics_collector::ReceivedMessageAgeCollector;
+using libstatistics_collector::topic_statistics_collector::ReceivedMessagePeriodCollector;
+namespace constants =
+  libstatistics_collector::topic_statistics_collector::topic_statistics_constants;
+
+/**
+ * Build a positive integer range to use in parameter descriptors.
+ *
+ * @return a positive integer range with specififies start, end and step values
+ */
+inline rcl_interfaces::msg::IntegerRange buildPositiveIntegerRange(
```

Comment:
nit: i'm inclined to suggest that these builder utilities go into a separate `msg_utils.hpp` with its own compilation unit, I don't know how much the `inline` really gains you in terms of performance. If the logic is really specific to the class, then maybe private members rather than global functions. But they do at a glance look like generic constructors for rcl_interfaces messages.